src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -105,12 +105,13 @@ object RuleUtils {
         }
       assert(filesByRelations.length == 1)
       indexes.filter(index =>
-        index.created && isHybridScanCandidate(index, filesByRelations.flatten))
+        index.created && index.deletedFiles.isEmpty && index.appendedFiles.isEmpty &&
```

Comment:
why is isEmpty checks needed for hybrid scan?

Reply:
Because entry.allSourceFiles (source.Content) includes the appended files - so need to handle the appended files properly.

Reply:
Is this an existing bug then? I wasn't sure why this is related to the config change.

Reply:
also, shouldn't we support nonEmpty case since this is hybrid scan?

Reply:
Yes that's a bug. It's from non-empty appendedFiles or deletedFile, not the config.

Reply:
Can you update the PR description please?

Reply:
and can you add a test that fixes the bug?

Reply:
We can support but need some change - but there is no agreement on this. Do you think this also should be delivered in 0.4?

Reply:
Yes, every bug fix should come with a test coverage. You can separate out the bug fix as a separate PR if you want (and just concentrate on the config changes in this PR).

Reply:
Actually, can you just separate out the bug fix as a separate PR?

Reply:
Discussed this offline, and we will add comment around this change.